### PR TITLE
[EC-318] Added some comments inside CORE pipelines

### PR DESCRIPTION
**Pull request by mamu0**
### List of changes

Comments have been added within the GitHub Actions "CORE" to make their use clearer.

### Motivation and context

It is used to specify when a Legacy pipe is needed and when one that uses the new infrastructure.

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: .github/workflows/core_code_review.yaml [hunk 1]
```diff
@@ -1,5 +1,12 @@
 name: PR - Core TF Validation
 
+# This pipeline start automatically when a PR is opened
```
Comment (gunzip) [inline .github/workflows/core_code_review.yaml:3]: # This pipeline starts automatically when a PR is opened.

It is responsible for managing changes related solely to the NEW infrastructure. Therefore, it checks whether the changes have occurred only in the directories listed in "paths."

## NOTE: 'NEW infrastructure' refers to the new Terraform infrastructure located in the src/core/prod folder, which no longer requires the terraform.sh script to be applied.

File: .github/workflows/core_code_review_weu.yaml [hunk 2]
```diff
@@ -1,5 +1,10 @@
 name: WEU - Core TF Validation (PROD)
 
+# This pipeline start automatically when a PR is opened
```
Comment (gunzip) [inline .github/workflows/core_code_review_weu.yaml:3]: # This pipeline starts automatically when a PR is opened.

It is responsible for managing only the legacy part (West Europe), ensuring that changes have occurred exclusively within the directories listed in "paths." It ignores the folders related to the new mode, indicated by the symbol '!', which denotes an excluded path.

File: .github/workflows/core_deploy.yaml [hunk 3]
```diff
@@ -1,5 +1,14 @@
 name: PR - Core TF Apply
 
+# This pipeline start automatically when a PR will be merged in 'main'
```
Comment (gunzip) [inline .github/workflows/core_deploy.yaml:3]: # This pipeline starts automatically when a PR is merged into 'main'.

It is responsible for managing changes related solely to the NEW infrastructure, ensuring that changes have occurred exclusively within the directories listed in "paths."

## NOTE: 'NEW infrastructure' refers to the new Terraform infrastructure located in the src/core/prod folder, which no longer requires the terraform.sh script to be applied.

This pipeline first executes a plan and then an apply, which must be approved by the team.

File: .github/workflows/core_deploy_weu.yaml [hunk 4]
```diff
@@ -1,6 +1,12 @@
 name: WEU - Core TF Apply (PROD)
 
-# Only manual
+# This pipeline start automatically when a PR will be merged in 'main'
```
Comment (gunzip) [inline .github/workflows/core_deploy_weu.yaml:3]: # This pipeline starts automatically when a PR is merged into 'main'.

It is responsible for managing only the legacy part (West Europe), ensuring that changes have occurred exclusively within the directories listed in "paths." It ignores the folders related to the new mode, indicated by the symbol '!', which denotes an excluded path.

This pipeline will use the terraform.sh script, which first executes a plan and then an apply, both of which must be approved by the team.

**Review by gunzip — CHANGES_REQUESTED** [review]
I've passed comments through AI

**Review by gunzip — APPROVED** [review]

**Review by christian-calabrese — APPROVED** [review]
LGTM

**Pull request merged**